thunderbirdparser/src/org/sleuthkit/autopsy/thunderbirdparser/MboxParser.java

```diff
@@ -97,63 +95,76 @@
     static boolean isValidMimeTypeMbox(byte[] buffer) {
         return (new String(buffer)).startsWith("From "); //NON-NLS
     }
-
+    
     /**
-     * Parse the mbox file and get the email messages.
-     *
-     * @param mboxFile
-     *
-     * @return a list of the email messages in the mbox file.
+     * Returns an instance of MBoxParser that will iterate and return EMailMessage
+     * objects with only the information needed for threading emails.
+     * 
+     * @param localPath String path to the mboxFile
+     * @param mboxFile The mboxFile to parse
+     * @return Instance of MboxParser
      */
-    List<EmailMessage> parse(File mboxFile, long fileID) {
+    static MboxParser getThreadInfoIterator(String localPath, File mboxFile) {
+        MboxParser parser = new MboxParser(localPath);
+        parser.createIterator(mboxFile, 0, false);
+        return parser;
+    }
+    
+    /**
+     * Returns an instance of MBoxParser that will iterate "whole" EmailMessages.
+     * 
+     * @param localPath String path to the mboxFile
+     * @param mboxFile The mboxFile to parse
+     * @param fileID The fileID of the abstractFile that mboxFile was found
+     * @return Instance of MboxParser
+     */
+    static MboxParser getEmailIterator(String localPath, File mboxFile, long fileID) {
+        MboxParser parser = new MboxParser(localPath);
+        parser.createIterator(mboxFile, fileID, true);
+        
+        return parser;
+    }
+    
+    /**
+     * Creates the real Iterator object instance.
+     * 
+     * @param mboxFile  The mboxFile to parse
+     * @param fileID    The fileID of the abstractFile that mboxFile was found
+     * @param wholeMsg  True if EmailMessage should have the whole message, 
+     *                  not just the thread information.
+     */
+    private void createIterator(File mboxFile, long fileID, boolean wholeMsg) {
         // Detect possible charsets
         List<CharsetEncoder> encoders = getPossibleEncoders(mboxFile);
-
-        CharsetEncoder theEncoder = null;
-        Iterable<CharBufferWrapper> mboxIterator = null;
+        
         // Loop through the possible encoders and find the first one that works.
         // That will usually be one of the first ones.
         for (CharsetEncoder encoder : encoders) {
             try {
-                mboxIterator = MboxIterator.fromFile(mboxFile).charset(encoder.charset()).build();
-                theEncoder = encoder;
+                Iterable<CharBufferWrapper> mboxIterable = MboxIterator.fromFile(mboxFile).charset(encoder.charset()).build();
+                if(mboxIterable != null) {
```

Comment:
I prefer a whitespace between the 'if' and the opening '('.  I think most of our existing code follows that convention.

I'm not sure if our coding guidelines specify it explicitly. 

There's many instances of "if(" and a few of "while(" in this PR.  I'll let @rcordovano decide if they should be changed in this PR and provide guidance of future.

Reply:
I'll just fix it.